language/move-prover/spec-lang/src/translate.rs

```diff
@@ -1536,8 +1553,14 @@ impl<'env, 'translator> ModuleTranslator<'env, 'translator> {
         let body = if self.spec_funs[spec_fun_idx].body.is_some() {
             std::mem::replace(&mut self.spec_funs[spec_fun_idx].body, None).unwrap()
         } else {
-            // No function body: the move function is already impure so no need to compute.
-            return false;
+            // If the function is native, consider it pure.
+            // Otherwise the function is non-native, its body cannot be parsed
+            // so we consider it impure.
+            // TODO(emmazzz) right now all the native move functions are considered pure.
```

Comment:
Doesn't that break already now with some of the vector functions which take `&mut` parameters? We really want to avoid that user would learn this via crashes/boogie errors. Perhaps you can check the signature of the Move function whether it contains any `&mut`?

Reply:
Right that's a good point. We should disallow them at least for now.